components/brave_wallet/browser/keyring_service.cc

```diff
@@ -2494,7 +2494,7 @@ KeyringService::GetZCashAddresses(const mojom::AccountIdPtr& account_id) {
        ++i) {
     auto key_id = mojom::ZCashKeyId::New(account_id->bitcoin_account_index,
                                          kBitcoinReceiveIndex, i);
-    auto address = zcash_keyring->GetAddress(*key_id);
+    auto address = zcash_keyring->GetTransparentAddress(*key_id);
```

Comment:
per https://github.com/brave/brave-core/pull/22870/files#r1567700571
This GetTransparentAddress should be hidden within ZcashKeyring.
HDKeyring::GetAddress() will call GetAddressInternal which each coin implemented , in this case it calculate transparent address.

Reply:
ZCash has a number of different address types and combinations - in the current release we support only transparent addresses but there is also orchard address and unified (orchard + transparent) address for the account.
This is done to explicitly show which address is actually used in order to reduce implicitness.

Reply:
We can't use GetAddress(size_t index) anyway since index is not enough to resolve address. Same approach in Bitcoin - there is additional  GetAddress(uint32_t account, const mojom::BitcoinKeyId& key_id) in the bitcoin_keyring.

Reply:
That interface is gone after this [refactor](https://github.com/brave/brave-core/pull/23075)